### PR TITLE
feat: Add logging configuration

### PR DESCRIPTION
**Pull request by msbrogli**
```bash
$ npm start

> hathor-wallet-headless@0.6.0 start /Users/msbrogli/Hathor/hathor-wallet-headless
> babel-node index.js

2021-02-26T06:23:39.033Z [wallet] info: Starting Hathor Wallet... {"wallet":"0.6.0","version":"v12.18.0","platform":"darwin","pid":15931}
2021-02-26T06:23:39.035Z [wallet] info: Set GAP LIMIT to 20
2021-02-26T06:23:39.041Z [wallet] info: Listening on localhost:8000...
2021-02-26T06:23:46.593Z [wallet] info: Wallet started with wallet id 123. Full-node info:  {"version":"0.38.0","network":"mainnet","min_weight":14,"min_tx_weight":14,"min_tx_weight_coefficient":1.6,"min_tx_weight_k":100,"token_deposit_percentage":0.01,"reward_spend_min_blocks":300,"max_number_inputs":255,"max_number_outputs":255}
2021-02-26T06:23:46.596Z [http] info: 127.0.0.1 - - [26/Feb/2021:06:23:46 +0000] "POST /start HTTP/1.1" 200 16 "-" "curl/7.64.1"
2021-02-26T06:23:51.598Z [wallet] debug: /send-tx failed {"body":{"outputs":[{"address":"H9YHCqJNfep3VBMMPZBAL7w1wh9ztRRbGo","value":100},{"address":"HN152wFDgy7ZopSf6PpSFMt8M8r2Mz9fGc","value":200}],"debug":true},"response":{"success":false,"error":"Token HTR: Insufficient amount of tokens","debug":{"balance":{"available":0,"locked":0},"partialData":{"tokens":[],"inputs":[],"outputs":[{"address":"H9YHCqJNfep3VBMMPZBAL7w1wh9ztRRbGo","value":100,"tokenData":0},{"address":"HN152wFDgy7ZopSf6PpSFMt8M8r2Mz9fGc","value":200,"tokenData":0}]},"txToken":{"name":"Hathor","symbol":"HTR","uid":"00"},"inputsAmount":0,"inputsLength":0,"outputsAmount":300}}}
2021-02-26T06:23:51.599Z [http] info: 127.0.0.1 - - [26/Feb/2021:06:23:51 +0000] "POST /wallet/send-tx HTTP/1.1" 200 425 "-" "curl/7.64.1"
```

## debug.log

```json
{"service":"wallet","wallet":"0.6.0","version":"v12.18.0","platform":"darwin","pid":15931,"level":"info","message":"Starting Hathor Wallet...","timestamp":"2021-02-26T06:23:39.033Z"}
{"message":"Set GAP LIMIT to 20","level":"info","service":"wallet","timestamp":"2021-02-26T06:23:39.035Z"}
{"message":"Listening on localhost:8000...","level":"info","service":"wallet","timestamp":"2021-02-26T06:23:39.041Z"}
{"service":"wallet","version":"0.38.0","network":"mainnet","min_weight":14,"min_tx_weight":14,"min_tx_weight_coefficient":1.6,"min_tx_weight_k":100,"token_deposit_percentage":0.01,"reward_spend_min_blocks":300,"max_number_inputs":255,"max_number_outputs":255,"level":"info","message":"Wallet started with wallet id 123. Full-node info: ","timestamp":"2021-02-26T06:23:46.593Z"}
{"service":"http","level":"info","message":"127.0.0.1 - - [26/Feb/2021:06:23:46 +0000] \"POST /start HTTP/1.1\" 200 16 \"-\" \"curl/7.64.1\"","timestamp":"2021-02-26T06:23:46.596Z"}
{"service":"wallet","body":{"outputs":[{"address":"H9YHCqJNfep3VBMMPZBAL7w1wh9ztRRbGo","value":100},{"address":"HN152wFDgy7ZopSf6PpSFMt8M8r2Mz9fGc","value":200}],"debug":true},"response":{"success":false,"error":"Token HTR: Insufficient amount of tokens","debug":{"balance":{"available":0,"locked":0},"partialData":{"tokens":[],"inputs":[],"outputs":[{"address":"H9YHCqJNfep3VBMMPZBAL7w1wh9ztRRbGo","value":100,"tokenData":0},{"address":"HN152wFDgy7ZopSf6PpSFMt8M8r2Mz9fGc","value":200,"tokenData":0}]},"txToken":{"name":"Hathor","symbol":"HTR","uid":"00"},"inputsAmount":0,"inputsLength":0,"outputsAmount":300}},"level":"debug","message":"/send-tx failed","timestamp":"2021-02-26T06:23:51.598Z"}
{"service":"http","level":"info","message":"127.0.0.1 - - [26/Feb/2021:06:23:51 +0000] \"POST /wallet/send-tx HTTP/1.1\" 200 425 \"-\" \"curl/7.64.1\"","timestamp":"2021-02-26T06:23:51.599Z"}
```

### CODE REVIEW

**Review by jansegre — APPROVED** [review]
Looks great!

File: config.js.template [hunk 1]
```diff
@@ -15,13 +15,27 @@ module.exports = {
       default: '',
   },
 
+  // Console logging level (default: info).
```
Comment (pedroferreira1) [inline config.js.template:18]: Should we list the options here?

File: index.js [hunk 2]
```diff
@@ -142,8 +146,6 @@ walletRouter.use((req, res, next) => {
     return;
   }
 
-  console.log('Received request to', req.originalUrl)
```
Comment (pedroferreira1) [inline index.js:145]: Why don't we change this to a `logger.debug`? I think it's helpful to log the requests the wallet headless is receiving.

File: index.js [hunk 3]
```diff
@@ -438,9 +448,16 @@ if (config.http_api_key) {
 app.use('/wallet', walletRouter);
 
 app.listen(config.http_port, config.http_bind_address, () => {
-  console.log(`Hathor Wallet listening on ${config.http_bind_address}:${config.http_port}...`);
+  console.log(`Listening on ${config.http_bind_address}:${config.http_port}...`);
```
Comment (pedroferreira1) [inline index.js:451]: I am thinking about logging more about the config. The gap limit is being logged when we set it but we could log the `tokenUid` and `connectionTimeout`. What do you think?

**Review by pedroferreira1 — APPROVED** [review]

File: index.js [hunk 4]
```diff
@@ -273,13 +275,21 @@ walletRouter.post('/send-tx', (req, res) => {
...
+        logger.debug('/send-tx failed', {
+          body: req.body,
+          response: response,
+        });
       }
       res.send(response);
     });
   } else {
     const response = {success: false, error: ret.message};
     if (debug) {
```
Comment (thr0wn) [inline index.js:287]: Is `if (debug)` required? I suppose `winston` should automatically ignore some logs based on the `loglevel` defined [here](https://github.com/HathorNetwork/hathor-wallet-headless/pull/54/files#diff-4e4bb3f3b1a497e10bb5d503f3ccb5072a1c0540e0020a00fa20da2cfd3ba2b3R26) and passed to `winston` [here](https://github.com/HathorNetwork/hathor-wallet-headless/pull/54/files#diff-ed4a110a094ac014f4bea8d6d93719d89d5e5a2d1974dc2f79e41faf30af470aR22).

**Review by thr0wn — CHANGES_REQUESTED** [review]

**Review by thr0wn — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (jansegre) [reply index.js:451]: We could probably do this structured, like `console.log(`Listening on ${config.http_bind_address}:${config.http_port}...`, config);` would log the whole config, but maybe sanitize sensitive info from config if there is any.

File: index.js (hunk 2 above)
Comment (msbrogli) [reply index.js:145]: I removed this log because we are using `morgan` to log all requests.

File: index.js (hunk 3 above)
Comment (msbrogli) [reply index.js:451]: I prefer to put these piece of information in a separate log. I'll do it and you can review again.

File: index.js (hunk 4 above)
Comment (msbrogli) [reply index.js:287]: This `debug` is related to the HTTP Request (and not to `winston`). It means the transaction is failing and the caller would like to get debugging information to find out what's wrong.

File: config.js.template (hunk 1 above)
Comment (msbrogli) [reply config.js.template:18]: Done! Thanks!

File: index.js (hunk 3 above)
Comment (msbrogli) [reply index.js]: Done!

**Review by jansegre — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (jansegre) [reply index.js]: Looks great!

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**